src/directives/formly-form.test.js

```diff
@@ -536,26 +536,85 @@ describe('formly-form', () => {
-      compileAndDigest()
-      expect(scope.fields[1].hide).be.false
-      scope.fields[0].formControl.$setViewValue('value')
-      expect(scope.fields[1].hide).be.true
+        it('should be called and resolve to true when field model changes', () => {
+          compileAndDigest()
+          expect(scope.fields[1].hide).be.false
+          scope.fields[0].formControl.$setViewValue('value')
+          expect(scope.fields[1].hide).be.true
+        })
+
+        it('should be called and resolve to false when field model changes', () => {
+          scope.fieldModel.foo = 'value'
+          compileAndDigest()
+          expect(scope.fields[1].hide).be.true
+          scope.fields[0].formControl.$setViewValue('')
+          expect(scope.fields[1].hide).be.false
+        })
+      })
+      describe('when a function is passed to hideExpression', () => {
+        beforeEach(() => {
+          scope.fields = [
+            {template: input, key: 'foo', model: scope.fieldModel},
+            {
+              template: input, key: 'bar',
+              model: scope.fieldModel,
+              hideExpression: ($viewValue, $modelValue, scope) => {
+                return !!scope.fields[1].data.formScope.fieldModel.foo
+              },
+              data: {
+                formScope: scope,
+              },
+            },
+          ]
+        })
+
+        it('should be called and resolve to true when field model changes', () => {
+          compileAndDigest()
+          expect(scope.fields[1].hide).be.false
+          scope.fields[0].formControl.$setViewValue('value')
+          expect(scope.fields[1].hide).be.true
+        })
+
+        it('should be called and resolve to false when field model changes', () => {
+          scope.fieldModel.foo = 'value'
+          compileAndDigest()
+          expect(scope.fields[1].hide).be.true
+          scope.fields[0].formControl.$setViewValue('')
+          expect(scope.fields[1].hide).be.false
+        })
+      })
     })
 
-    it('should be called and resolve to false when field model changes', () => {
-      scope.fieldModel.foo = 'value'
+    it('ensures that hideExpression has all the expressionProperties values', () => {
+      scope.options = {formState: {}}
+      scope.fields = [{
+        template: input,
+        key: 'test',
+        hideExpression: `
+        options === options.data.field &&
+        index === 0 &&
+        formState === options.data.formOptions.formState &&
+        originalModel === options.data.originalModel &&
+        formOptions === options.data.formOptions`,
+        data: {
+          originalModel: scope.model,
+          formoptions: scope.options,
+        },
+      }]
+      scope.fields[0].data.field = scope.fields[0]
       compileAndDigest()
-      expect(scope.fields[1].hide).be.true
-      scope.fields[0].formControl.$setViewValue('')
-      expect(scope.fields[1].hide).be.false
+      expect(scope.fields[0].hide).be.false
```

Comment:
Line 617 had to be "expect(scope.fields[0].hide).be.true" instead of `false`, it escaped me somehow. I'm working on another commit right now.